PyInstaller/building/build_main.py

```diff
@@ -151,13 +154,37 @@ def find_binary_dependencies(binaries, binding_redirects, import_packages):
     if compat.is_ms_app_store:
         extra_libdirs.append(compat.base_prefix)
 
+    # On Windows with python >= 3.8, attempt to track calls to os.add_dll_directory() to obtain DLL search paths that
+    # are dynamically set during packages' initialization.
+    if compat.is_win and compat.is_py38:
+        added_dll_directories = []
+        _orig_add_dll_directory = os.add_dll_directory
+
+        def _pyi_add_dll_directory(path):
+            added_dll_directories.append(path)
+            return _orig_add_dll_directory(path)
+
+        os.add_dll_directory = _pyi_add_dll_directory
+
     # Import collected packages to set up environment
     for package in import_packages:
         try:
             __import__(package)
         except Exception:
             pass
 
+    # Process extra search paths on Windows
+    if compat.is_win:
+        # Directories added via os.add_dll_directory() calls.
+        if compat.is_py38:
+            logger.info("Extra DLL search directories (AddDllDirectory): %r", added_dll_directories)
+            extra_libdirs += added_dll_directories
+
+        # Directories set via PATH environment variable
+        path_directories = [directory for directory in os.environ.get("PATH", '').split(os.pathsep)]
```

Comment:
Do we not need something to keep out empty directory paths?
```suggestion
        path_directories = [directory for directory in os.environ.get("PATH", '').split(os.pathsep) if directory]
```

Reply:
Good point. Do you think we should be extra paranoid and also trim the entries?

Reply:
*Trim* meaning `dependency.strip()` I take it? I can't see how that would ever be needed?

Reply:
> Do you think we should be extra paranoid and also trim the entries?

Nevermind. Looks like leading or trailing whitespace renders that particular path invalid (at least within `cmd`), so it is probably not a case we should worry about.

But empty entries do need to be handled.